addons/account/models/account_move.py

```diff
@@ -3689,12 +3702,12 @@ class AccountMoveLine(models.Model):
         ),
         (
             'check_accountable_required_fields',
-             "CHECK(COALESCE(display_type IN ('line_section', 'line_note'), 'f') OR account_id IS NOT NULL)",
+             "CHECK(COALESCE(line_type IS NOT NULL AND line_type IN ('invl_section', 'invl_note'), 'f') OR account_id IS NOT NULL)",
```

Comment:
why do you need `line_type IS NOT NULL`

Reply:
This may be something I added while iterating and trying to fix tests. I'll remove them and check that everything is still ok